cpp/src/reductions/scan.cu

```diff
@@ -34,6 +33,32 @@
 
 namespace cudf {
 namespace detail {
+
+namespace {
+/**
+ * @brief Accessor handles both nullable and non-nullable columns.
+ *
+ * @tparam Element type used for null-replacement value
+ */
+template <typename Element>
+struct null_replace_accessor {
```

Comment:
We already have a null replacement iterator. How is this different?

Reply:
This one allows the column to not have a null-mask.
I think it should go into the `interator.cuh` but then I need to think of a name for it.

Reply:
Or perhaps I can change the current one to this one with appropriate defaults.